fs/dafilesrv/dafilesrv.cpp

```diff
@@ -386,7 +386,29 @@ int main(int argc, const char* argv[])
     StringBuffer componentName;
 
     // NB: bare-metal dafilesrv does not have a component specific xml
-    Owned<IPropertyTree> config = loadConfiguration(defaultYaml, argv, "dafilesrv", "DAFILESRV", nullptr, nullptr);
+    Owned<IPropertyTree> extractedGlobalConfig = createPTree("dafilesrv");
+
+#ifndef _CONTAINERIZED
+    Owned<IPropertyTree> env = getHPCCEnvironment();
+    IPropertyTree* globalTracing = env->getPropTree("Software/tracing");
+    if (globalTracing != nullptr)
+        extractedGlobalConfig->addPropTree("tracing", globalTracing);
+#endif
+
+    const char* componentTag = "dafilesrv";
+    Owned<IPropertyTree> componentDefault;
+    if (defaultYaml)
+    {
+        Owned<IPropertyTree> defaultConfig = createPTreeFromYAMLString(defaultYaml, 0, ptr_ignoreWhiteSpace, nullptr);
+        componentDefault.set(defaultConfig->queryPropTree(componentTag));
+        if (!componentDefault)
```

Comment:
this should never be true, the const defaultYaml explicitly defines "dafilesrv" so I don't think there's any point in checking (code currently leads me to think it may be missing)